### ISSUE TITLE
Support for full memory dumps from ESX 

### ISSUE THREAD

**Issue by marklech** [cuckoosandbox/cuckoo#975]
Hi, 
I noticted there's no support for full memory dumps 
of machines running on ESXi. Someone previosly 
posted an extentsion to the esx.py machinery code that 
implements this functionality on Cuckoo's community 
(by suspending a machine and accessing the datastore 
to download the image), but it's gone now. Is it possible
to add these feature to the current esx machinery implementation?
Regards, 
Mark.

**Comment by jbremer** [comment]
Do you still have a backup of said code? I can't do any ESXi related testing myself, but if one or two people acknowledge that it works I'm happy to merge it.

**Comment by marklech** [comment]
Unfortunately not. This is the original thread from the community: https://community.cuckoosandbox.org/t/full-memory-dumps-in-esx/174 
The pastebin put there is gone , I lost my backup of the code and I can't reach the guy who published it. 
If you could provide a basic version for this functionaIity I can make the necessary adjustments . If that is too much , I can attempt to do it on my own and contribute the result.

**Comment by jbremer** [comment]
I believe that was @certau-davidg, see also the following comment https://github.com/cuckoosandbox/cuckoo/issues/927#issuecomment-223189619. Hopefully he'll be able to share the patch again :-)

**Comment by certau-davidg** [comment]
As requested, grab it while it is hot:
[Pastebin esx.conf](http://pastebin.com/GfYAvbT5)
[Pastebin esx.py](http://pastebin.com/yWNLKHLa)

Feel free to improve it.

**Comment by marklech** [comment]
Thanks a lot guys ! I really appreciate the effort on the code and bringing it back here .
David, thanks for the response on the community issue. I will verify it soon .

**Comment by jbremer** [comment]
Ehm, that commit was meant for #976.

**Comment by jgajek** [comment]
This is already supported in the Vsphere machinery module.

**Comment by certau-davidg** [comment]
While that is true for some builds, you still need to buy vsphere for that.
The esxi module allows no-frills-cheapo cuckoo platforms.

**Comment by jgajek** [comment]
No, you can run the vSphere machinery module on standalone ESXi hosts.

**Comment by certau-davidg** [comment]
Hmm ok, we honestly missed that addition when we migrated our esxi fix over to 2.0-rc1.
Does that mean that the esxi module is essentially deprecated for esxi 5.5/6.0?

**Comment by jgajek** [comment]
I don't think the libvirt driver supports ESXi 6.x, so I'm not sure if it's even an option right now.

**Comment by RedWolf74** [comment]
As much as I got from the docs, libvirt doesn't support memory download for ESX(i) at all. The vSpehere machinery module works well for stand alone ESXi and thanks to jgajek, now for vCenter as well. So from my point of view, the esx module can be called deprecated.

**Comment by KillerInstinct** [comment]
I've been saying the esx module could be depricated ever since jgajek released his vSphere module. The only down side to depricating the esx module is the manditory pyvmomi library installation. But that's been in pypi for ages now -- not a big deal in my book.

**Comment by jbremer** [comment]
If all parties agree I'm fine with removing the `esx` machinery module.

**Comment by botherder** [comment]
Frankly, I have little to no experience with ESX/vSphere, so I don't really know how to weigh in on this.
